server/src-lib/Hasura/RQL/DDL/RemoteSchema.hs

```diff
@@ -156,3 +195,176 @@ fetchRemoteSchemas =
      |] () True
   where
     fromRow (n, Q.AltJ def, comm) = AddRemoteSchemaQuery n def comm
+
+runAddRemoteSchemaPermissions
+  :: ( QErrM m, UserInfoM m
+     , CacheRWM m, MonadTx m
+     )
+  => RemoteSchemaPermissions -> m EncJSON
+runAddRemoteSchemaPermissions q = do
+  adminOnly
+  newRSCtx <- runAddRemoteSchemaPermissionsP1 q
+  runAddRemoteSchemaPermissionsP2 q newRSCtx
+  pure successMsg
+
+runAddRemoteSchemaPermissionsP1
+  :: (QErrM m, CacheRM m) => RemoteSchemaPermissions -> m RemoteSchemaCtx
+runAddRemoteSchemaPermissionsP1 remoteSchemaPermission = do
+  validateRemoteSchemaPermissions remoteSchemaPermission
```

Comment:
Why another function?

Reply:
because it's clear that P1 only does validate.